docker.go

```diff
@@ -79,16 +94,36 @@ func (container *Container) Volume(hostPath, containerPath, mode string) *Contai
 func NewDockerContainer() *Container {
 	container := new(Container)
 	container.Type = "DOCKER"
-	container.Docker = &Docker{
-		Image:        "",
-		Network:      "BRIDGE",
-		PortMappings: make([]*PortMapping, 0),
-		Parameters:   make([]*Parameters, 0),
-	}
-	container.Volumes = make([]*Volume, 0)
+	container.EmptyVolumes()
+	container.Docker = &Docker{}
+	container.Docker.Container("").
+		Bridged().
+		SetPrivileged(false).
```

Comment:
Not certain that I understand you completely here, partly because the term _empty_ is a bit ambiguous in our context. (Does it mean omitting as in the resulting JSON data, or zero-valued?)

Your first use case seems to be that of setting all values to zero, which doesn't sound like a common case as there's no inherent semantics in "everything zero-valued". The more common case to me is "I don't care about any value except for the following".

Having to fetch an application configuration prior to making changes to individual fields is at least theoretically prone to race conditions. Additionally, I'd rather not want go-marathon to reflect Marathon's default settings as these may be subject to change, causing things to break from one version to the next in the worst case. Finally, Marathon's `PUT` already has change-the-given-parameters-only semantics, so I'd prefer to use the existing mechanism.

That being said, I happen to realize now that things would be easier if the Marathon REST API had separate replace and update functionalities (i.e., real `PUT` and `PATCH`).